src/elliptic_curve.rs

```diff
@@ -54,92 +48,72 @@ impl<F: Field + Clone + PartialEq> EllipticCurve<F> {
         // y = ( - b + sqrt( delta ) ) / 2
         let rand_y = half.mul(&b.clone().neg().add(&delta.clone().sqrt()));
 
-        ECPoint::RATIONALPOINT(RationalPoint {
-            curve: self,
-            coord: (rand_x, rand_y),
-        })
+        ECPoint::AffinePoint(rand_x, rand_y)
```

Comment:
Since curve is not in the struct anymore, maybe we need to add a check to ensure point in ob the curve when we work with them